### PR TITLE
Trajopt fixes

### PR DESCRIPTION
**Pull request by simonschmeisser**
These are tiny fixes to the trajopt plugin that allow it to work in obstacle-free cases when direct joint interpolation is a valid solution.

### CODE REVIEW

**Review by JafarAbdi — APPROVED** [review]
lgtm

File: moveit_planners/trajopt/src/trajopt_interface.cpp [hunk 1]
```diff
@@ -166,7 +168,7 @@ bool TrajOptInterface::solve(const planning_scene::PlanningSceneConstPtr& planni
     res.error_code.val = moveit_msgs::MoveItErrorCodes::INVALID_GOAL_CONSTRAINTS;
     return false;
   }
-  else if (!req.goal_constraints[0].orientation_constraints.empty() &&
+  else if (!req.goal_constraints[0].position_constraints.empty() &&
            req.goal_constraints[0].orientation_constraints.empty())
   {
     ROS_ERROR_STREAM_NAMED("trajopt_planner", "orientation constraint is not defined");
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_interface.cpp:174]:
```suggestion
    ROS_ERROR_STREAM_NAMED("trajopt_planner", "Both position and orientation constraints are not defined");
```

**Comment by simonschmeisser** [comment]
> Are you working on TrajOpt? I'm planning to continue the port on my spare time, do you want to meet to discuss it?

I haven't managed to include it in my work time yet so it's spare time progress as well ... I have another branch where I copied and rebased the collision term from Omids branch to current upstream trajopt and it compiles but I haven't found the time to actually test it yet ... switching the collision checker is annoyingly complicated in MoveIt ...

**Comment by JafarAbdi** [comment]
> > Are you working on TrajOpt? I'm planning to continue the port on my spare time, do you want to meet to discuss it?
> 
> I haven't managed to include it in my work time yet so it's spare time progress as well ... I have another branch where I copied and rebased the collision term from Omids branch to current upstream trajopt and it compiles but I haven't found the time to actually test it yet ... switching the collision checker is annoyingly complicated in MoveIt ...

Is the branch public? Do you mind sharing it?

**Comment by simonschmeisser** [comment]
> > > Are you working on TrajOpt? I'm planning to continue the port on my spare time, do you want to meet to discuss it?
> > 
> > 
> > I haven't managed to include it in my work time yet so it's spare time progress as well ... I have another branch where I copied and rebased the collision term from Omids branch to current upstream trajopt and it compiles but I haven't found the time to actually test it yet ... switching the collision checker is annoyingly complicated in MoveIt ...
> 
> Is the branch public? Do you mind sharing it?

Please see the completely untested #3235

**Pull request merged**